src/Directive.php

```diff
@@ -29,10 +32,27 @@ abstract class Directive
     public const WAKE_LOCK = 'wake-lock';
     public const XR = 'vr';
 
-    public static function isValid(string $directive): bool
+    public static function make(string $directive, string $type = 'default'): DirectiveContract
+    {
+        return match ($type) {
+            'default' => DefaultDirective::directive($directive),
+            default => throw InvalidDirective::unknownDirectiveType($type),
+        };
+    }
+
+    public function addRule(string $rule): static
     {
-        $constants = (new ReflectionClass(static::class))->getConstants();
+        if (in_array($rule, $this->rules, true)) {
+            return $this;
+        }
 
-        return in_array($directive, $constants);
+        $this->rules[] = $rule;
+
+        return $this;
+    }
+
+    public function rules(): array
```

Comment:
Get all the rules.

I'm using a public method to get the contents of the property instead of making the property public.
Reasoning behind this is that if the property is public that we can't do any checks on what is put inside it. Possibly duplicate rules could be added. This now won't happen because we check for this in the `addRule(..)` method.